### PR TITLE
Allow reading `bool` as integer, like `MySql.Data` allows (Fixes #782)

### LINKED ISSUE #782: Can't convert Bool to Int32 (TINYINT)
With Oracle's `MySql.Data`, when you read a TINYINT column - you can read it as a `bool` or as an `Int32` - either way works.

And it only makes sense as `TINYINT` is not explicitly a boolean, it's just so tiny that it can only hold 0 or 1... And it's called TINY**INT**.

Seems like an easy fix.

It currently throws: `Can't convert Bool to Int32`
From: `MySqlConnector\Core\Row.cs:line 224`

For me - this was an unexpected breaking change when trying the transition to this library.

### PR DESCRIPTION
**Pull request by danielgindi**

### CODE REVIEW

File: src/MySqlConnector/Core/Row.cs [hunk 1]
```diff
@@ -207,16 +209,15 @@ public int GetInt32(int ordinal)
...
 				columnDefinition.ColumnType != ColumnType.Short &&
 				columnDefinition.ColumnType != ColumnType.Int24 &&
 				columnDefinition.ColumnType != ColumnType.Long &&
 				columnDefinition.ColumnType != ColumnType.Longlong &&
 				columnDefinition.ColumnType != ColumnType.Bit &&
 				columnDefinition.ColumnType != ColumnType.Year &&
 				columnDefinition.ColumnType != ColumnType.Decimal &&
-				columnDefinition.ColumnType != ColumnType.NewDecimal) ||
-				(columnDefinition.ColumnType == ColumnType.Tiny && Connection.TreatTinyAsBoolean && columnDefinition.ColumnLength == 1 && (columnDefinition.ColumnFlags & ColumnFlags.Unsigned) == 0))
+				columnDefinition.ColumnType != ColumnType.NewDecimal)
```
Comment (bgrainger) [inline src/MySqlConnector/Core/Row.cs]: I suspect this fails to coerce "bool" values to 0 or 1.

Comment (bgrainger) [reply src/MySqlConnector/Core/Row.cs]: Fixed: 693b8e0b54fad79ff076b85ae44f34b9d1042428

**Comment by danielgindi** [comment]
Great!

Btw how do you explain why GetInt32 is so different the implementation of other int getters?
It seems like it’s duplicating the code of GetValue, adding a lot of checks and manually parsing the raw data.

**Comment by bgrainger** [comment]
See https://github.com/mysql-net/MySqlConnector/issues/721 for the rationale.

**Pull request merged**